autoload/ctrlp/imports.vim

```diff
@@ -0,0 +1,82 @@
+let s:go_imports_var = {
+      \  'init':   'ctrlp#imports#init()',
+      \  'exit':   'ctrlp#imports#exit()',
+      \  'enter':  'ctrlp#imports#enter()',
+      \  'accept': 'ctrlp#imports#accept',
+      \  'lname':  'go imports',
+      \  'sname':  'imports',
+      \  'type':   'tabs',
+      \}
+
+if exists('g:ctrlp_ext_vars') && !empty(g:ctrlp_ext_vars)
+  let g:ctrlp_ext_vars = add(g:ctrlp_ext_vars, s:go_imports_var)
+else
+  let g:ctrlp_ext_vars = [s:go_imports_var]
+endif
+
+function! ctrlp#imports#init()
+  cal s:enable_syntax()
+  return s:imports
+endfunction
+
+function! ctrlp#imports#exit()
+  unlet! s:imports
+endfunction
+
+function! ctrlp#imports#accept(mode, str)
+  call ctrlp#exit()
+  call go#import#SwitchImport(1, '', split(a:str, "\t", 1)[0], '')
+endfunction
+
+function! ctrlp#imports#enter()
+  let s:imports = []
+
+  let bin_path = go#path#CheckBinPath('go')
+  if empty(bin_path)
+    return
+  endif
+
+  " get a list of all available imports including package docs
+  let command = printf("%s list -f \"{{.ImportPath}}\t{{.Doc}}\" ...", bin_path)
+
+  let out = go#util#System(command)
+  if go#util#ShellError() != 0
+    call go#util#EchoError(out)
```

Comment:
This is not good. For example this doesn't work for me because in my GOPATH there are files that don't get compiled. But because the command above calls `...` it searches all packages under my GOPATH. It fails. And when you echo all these, my whole Vim screen get black. I think we should come up with a faster and better solution. Did you think to implement it as a part of `motion`? https://github.com/fatih/motion 

I dont want to be look picky but I've explicitly said https://github.com/fatih/vim-go/blob/master/.github/CONTRIBUTING.md to open an issue before opening PR's like these. I like the idea but it's something we need to talk before :)